### PR TITLE
Samples - Add and change preloaded samples

### PR DESCRIPTION
**Pull request by Ark-kun**
Add the "Data passing in python components" tutorial and the combined "DSL control structures" tutorial.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2944)

### CODE REVIEW

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    print_text(numbers_task.output)
+
+    sum_task = sum_numbers(numbers_task.outputs['numbers'])
+    print_text(sum_task.output)
+
+
+# Submit the pipeline for execution:
+#kfp.Client(host=kfp_endpoint).create_run_from_pipeline_func(sum_pipeline, arguments={})
+
+# Combining all pipelines together in a single pipeline
```
Comment (numerology) [inline samples/tutorials/Data passing in python components/Data passing in python components - Files.py]: Shall we call out this is a way to achieve 'composable pipeline'?

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
+# %% [markdown]
```
Comment (numerology) [inline samples/tutorials/Data passing in python components/Data passing in python components - Files.py]: I'm curious how can one open the .py file and apply this md magic can be properly rendered?

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py (hunk 1 above)
Comment (Ark-kun) [reply samples/tutorials/Data passing in python components/Data passing in python components - Files.py]: This sample was created from a notebook which had ~10 pipelines. But a preloaded sample is only a single compiled pipeline file. So we're basically combining three unrelated pipelines in a single file.

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py (hunk 2 above)
Comment (Ark-kun) [reply samples/tutorials/Data passing in python components/Data passing in python components - Files.py]: VS Code supports converting both ways between Jupyter and Python: https://code.visualstudio.com/docs/python/jupyter-support-py#_export-a-jupyter-notebook

P.S. If you think that these comment magics should not be in the file, I can remove them.

File: samples/tutorials/DSL - Control structures/DSL - Control structures.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# %% [markdown]
```
[the remark this answers is not in the document]
Comment (Ark-kun) [reply samples/tutorials/DSL - Control structures/DSL - Control structures.py]: Done

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py (hunk 2 above)
Comment (numerology) [reply samples/tutorials/Data passing in python components/Data passing in python components - Files.py]: No worry, you can leave as it is

**Comment by Bobgy** [comment]
Note, pls take a look at Dockerfile, it pins all urls with blob/master, but not tree/master. Maybe we should do both. I'm not sure what is difference in these

**Pull request merged**